flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/TableFactoryService.java

```diff
@@ -255,10 +256,11 @@
 	private static <T extends TableFactory> List<T> filterByContext(
 			Class<T> factoryClass,
 			Map<String, String> properties,
-			List<TableFactory> foundFactories,
 			List<T> classFactories) {
 
-		List<T> matchingFactories = classFactories.stream().filter(factory -> {
+		List<T> matchingFactories = new ArrayList<>();
+		Tuple3<T, Integer, Map<String, Tuple2<String, String>>> bestMatched = null;
```

Comment:
Could you create a inner class to encapsulate the matched information? It's really hard to guess what does every part mean.